cephbackupstorage/cephbackupstorage/cephagent.py

```
@@ -1072,6 +1149,562 @@ def cancel(self, req):
+                    raise Exception("scp timed out after 1200 seconds")
+                if r != 0:
+                    raise Exception("scp failed: %s" % e)
+
+            def create_remote_sudo_passwd_file():
+                logger.info("creating remote sudo passwd file on remote host")
+                # Use 'umask 077' so the file is created with 0600 permissions
+                # from the start, eliminating the TOCTOU race window that would
+                # exist with 'touch FILE && chmod 600 FILE'.
+                create_passwd_file_cmd = '{0} "umask 077 && cat > {1}"'.format(
+                    sshpass_cmd_header, quoted_remote_sudo_passwd_file_path)
+                passwd_process = subprocess.Popen(create_passwd_file_cmd, shell=True, stdin=subprocess.PIPE,
```

Ruff (0.15.10)

[error] `subprocess` call with `shell=True` identified, security issue

(S602)